### ISSUE TITLE
Moved Excetion

### ISSUE THREAD

**Issue by yuming19831007** [StackExchange/StackExchange.Redis#276]
string key = "name_RedisTest1_default";
            using (ConnectionMultiplexer redis = ConnectionMultiplexer.Connect("10.100.1.212"))
            {
                var db = redis.GetDatabase();
                db.StringSet(key, "fjahsfjkhasjdfhajdshfjkadshfajk");
            }
            for (int i = 0; i < 1000; i++)
            {
                using (ConnectionMultiplexer redis = ConnectionMultiplexer.Connect("10.100.1.212"))
                {
                    var db = redis.GetDatabase();
                    db.StringGet(key);
                    Console.WriteLine(i.ToString());
                }
            }
        

error:MOVED 6202 10.100.1.213:6379

**Comment by kylestittleburg** [comment]
I can reproduce this using one of our test clusters using the above code.  However, it does work consistently for me if I re-use a single `ConnectionMultiplexer` as the documentation recommends, if you're looking for a workaround.

```
            string key = "name_RedisTest1_default";
            using (ConnectionMultiplexer redis = ConnectionMultiplexer.Connect("10.1.13.63"))
            {
                var db = redis.GetDatabase();
                db.StringSet(key, "fjahsfjkhasjdfhajdshfjkadshfajk", TimeSpan.FromMinutes(1));

                for (int i = 0; i < 1000; i++)
                {
                    db = redis.GetDatabase();
                    db.StringGet(key);
                    Console.WriteLine(i.ToString());
                }
            }
```

**Comment by NickCraver** [comment]
This should be resolved in 1.1.553-alpha (latest), after #312 was merged in.

**Comment by NickCraver** [comment]
Merging into #248 for tracking of MOVED exceptions.
